cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -461,6 +706,8 @@ std::shared_ptr<ScalarFunction> MakeArithmeticFunction(std::string name,
     auto exec = ArithmeticExecFromOp<ScalarBinaryEqualTypes, Op>(ty);
     DCHECK_OK(func->AddKernel({ty, ty}, ty, exec));
   }
+  AddDecimalBinaryKernels<Op>(name, &func);
```

Comment:
Please ensure we're not adding decimal kernels for functions which don't support decimals yet:
```suggestion
  // TODO($FOLLOW_UP_JIRA) remove when decimal is supported for all arithmetic kernels
  if (name == "add" || name == "add_checked" || ...) {
    AddDecimalBinaryKernels<Op>(name, &func);
  }
```

Reply:
Please open a follow up JIRA for supporting decimals in the other arithmetic functions as well

Reply:
I moved `AddDecimalBinaryKernels` calls out of `MakeArithmeticFunction` and `MakeArithmeticFunctionNotNull`, and put it under each kernel supporting decimal operations.
https://github.com/apache/arrow/pull/10364/files#diff-3eafd7246f6a8c699f10d46e3276852fe44b6853b5517ef10396e561730c09f4R840

I'm afraid some arithmetic kernels (e.g., `power`) may not support decimals?

Reply:
Thanks, that will work

> I'm afraid some arithmetic kernels (e.g., power) may not support decimals?

This is one topic which the follow up JIRA would give us a better place to discuss